### PR TITLE
Lack of RateLimiting

### PR DESCRIPTION
**Pull request by B3EF**
## ✍️ Description
Encountered **Lack of RateLimiting** in the login page of octobercms/october repo.
bug files at folder october/2.
## 🕵️‍♂️ Proof of Concept
1. Clone and setup [octobercms](https://github.com/octobercms/october).
2. Lack of RateLimiting at http://localhost/octobercms/backend/backend 
3. Proof Pics:
    - [fig1](https://cdn.discordapp.com/attachments/749019614352244777/749022200907956305/unknown.png)
## 💥 Impact
Attacker is able to perform bruteforce attack.
## ✅ Checklist
- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Review by Mik317 — APPROVED** [review]
Hi @B3EF :smile:,
I installed the software and tried bruteforcing a user `admin` and it seems `rate limit` is implemented, in fact after 50 requests the `admin` account is blocked and you can't login again as him even with the correct password.

Screenshot from 2020-08-29 17-16-09

I was able to validate however the fact that, changing only the `username`, any protection isn't implemented to avoid `user enumeration` :smile:

Good findings :smile:

Cheers,
Mik

**Comment by Mik317** [comment]
@B3EF please let me know if I'm wrong.
I just saw you opened also the #438 PR, but in the `vulnerability.json` of this PR, the `summary` states `user enumeration`.

If this one was concerning only the absence of `rate limit` which permits `bruteforcing`, then this is **not valid** as per first screen.

Just let me know if I understood correctly and I'll `validate` the #438 and unvalidate this one :smile:

Cheers,
Mik

**Comment by B3EF** [comment]
> @B3EF please let me know if I'm wrong.
> I just saw you opened also the #438 PR, but in the `vulnerability.json` of this PR, the `summary` states `user enumeration`.
> 
> If this one was concerning only the absence of `rate limit` which permits `bruteforcing`, then this is **not valid** as per first screen.
> 
> Just let me know if I understood correctly and I'll `validate` the #438 and unvalidate this one 😄
> 
> Cheers,
> Mik
that one was actually the one you should validate .<3

File: bounties/packagist/october/2/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Description
+Encountered **Lack of RateLimiting** in login page of octobercms/october repo.
```
Comment (Mik317) [inline bounties/packagist/october/2/README.md:2]: TEST: Add review to change status to `need more info` :smile:

**Review by Mik317 — CHANGES_REQUESTED** [review]

**Comment by Mik317** [comment]
@B3EF :smile:,
thanks for the clarification :+1: 

I've unvalidated this one and I'm going to validate #438 :100: 

Cheers,
Mik

**Pull request closed without merging**